File: Android/6.resource/app/src/main/java/fast/campus/myapplication/MainActivity.java
package fast.campus.myapplication;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    ImageView imageView;
    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //xml의  뷰 컴포넌트를 자바로 변경하는 방법
        imageView = findViewById(R.id.imageView);
        textView = findViewById(R.id.textView);

        //xml의  뷰 컴포넌트를 동적으로 변경하는 방법
        imageView.setImageResource(R.drawable.dog2);
        textView.setText("안녕하세요");

        //xml의  뷰 컴포넌트를 string으로 변경하는 방법
        String content = getApplicationContext().getResources().getString(R.string.long_text);
        textView.setText(content);

        getApplicationContext().getResources().getStringArray(R.array.string_array);
        Log.d("String_array",getApplicationContext().getResources().getStringArray(R.array.string_array)[1]);
        //
    }
}

File: Android/myappweather/app/src/main/java/weather/fastcampus/myapplication/MainActivity.kt
package weather.fastcampus.myapplication

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import com.firebase.ui.auth.AuthUI
import com.firebase.ui.auth.IdpResponse
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {

    private val RC_SIGN_IN = 1000

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        Log.d("start","MainActivity onCreate")

        // Choose authentication providers
        checkPreviousLogin()

//        val providers = arrayListOf(
//            AuthUI.IdpConfig.EmailBuilder().build(),
//            AuthUI.IdpConfig.GoogleBuilder().build()
//        )

        // Create and launch sign-in intent
//        startActivityForResult(
//            AuthUI.getInstance()
//                .createSignInIntentBuilder()
//                .setAvailableProviders(providers)
//                .setLogo(R.drawable.icons8_sun_64)
//                .setTheme(R.style.GreenTheme)
//                .build(),
//            RC_SIGN_IN
//        )

    }

    private fun checkPreviousLogin(){
        val user = FirebaseAuth.getInstance().currentUser
        if(user == null) showLoginWindow()
        else moveToOpenWeatherActivity()
    }

    private fun moveToOpenWeatherActivity(){
        startActivity(Intent(this, OpenWeatherActivity::class.java))
    }

    private fun showLoginWindow(){
        val providers = arrayListOf(
            AuthUI.IdpConfig.EmailBuilder().build(),
            AuthUI.IdpConfig.GoogleBuilder().build()
        )
        startActivityForResult(
            AuthUI.getInstance()
                .createSignInIntentBuilder()
                .setAvailableProviders(providers)
                .setLogo(R.drawable.icons8_sun_64)
                .setTheme(R.style.GreenTheme)
                .build(),
            RC_SIGN_IN
        )
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (requestCode == RC_SIGN_IN) {
            val response = IdpResponse.fromResultIntent(data)

            if (resultCode == Activity.RESULT_OK) {
                // Successfully signed in
                val user = FirebaseAuth.getInstance().currentUser
                // ...
                startActivity(Intent(this,OpenWeatherActivity::class.java))

            } else {
                // Sign in failed. If response is null the user canceled the
                // sign-in flow using the back button. Otherwise check
                // response.getError().getErrorCode() and handle the error.
                // ...
                Toast.makeText(this,"로그인 실패, 로그인을 다시 시도해주세요!",Toast.LENGTH_LONG).show()
            }
        }
    }
}
